app/stores/AppStore.js

```diff
@@ -32,23 +34,30 @@ export default class AppStore extends Store {
 
   @computed get currentLocale(): string {
     if (!this.isCurrentLocaleSet) {
+      this.getProfileLocaleRequest.execute();
       return 'en-US';
     }
-    return this.profileLocaleRequest.result;
+    return this.getProfileLocaleRequest.result || this.setProfileLocaleRequest.result;
+  }
+
+  @computed get hasLoadedCurrentLocale(): bool {
+    return this.getProfileLocaleRequest.wasExecuted;
   }
 
   @computed get isCurrentLocaleSet(): bool {
-    return this.profileLocaleRequest.wasExecuted;
+    if (this.getProfileLocaleRequest.result == null) return false;
+    return (this.getProfileLocaleRequest.wasExecuted && this.getProfileLocaleRequest.result) ||
+           (this.setProfileLocaleRequest.wasExecuted && this.setProfileLocaleRequest.result);
```

Comment:
True, you're right - there is an additional improvement you can make:

```
this.setProfileLocaleRequest.execute(locale).then(() => this.getProfileLocaleRequest.invalidate().patch(() => locale)
```
This patches the result of the `getProfileLocaleRequest` and marks it as dirty (so it is actually re-fetched the next time `execute` is called. But it avoids an immediate additional call 😉